symengine/cwrapper.cpp

```diff
@@ -170,34 +189,46 @@ TypeID basic_get_type(const basic s)
     return static_cast<TypeID>(s->m->get_type_code());
 }
 
-void symbol_set(basic s, const char *c)
+int symbol_set(basic s, const char *c)
```

Comment:
Regarding the output type, how about calling it `CWRAPPER_OUTPUT_TYPE` (perhaps you can find a better name that is not so long). 

This output variable (be it `void, `int` or `ExceptionType *` or `void *`) is only touched by the `CWRAPPER_END` macros. So if we instead use a `CWRAPPER_OUTPUT_TYPE` macro, then we have the freedom to change from `void` (the current master) to `int` or even to a pointer and allocate a struct with the exception info. All this by only modifying the macros, without touching the rest of the C wrappers.

What do you think?